test/upgrade/skip_at_verification_test.go

```diff
@@ -23,31 +23,31 @@ import (
 )
 
 func TestSkipAtBackgroundVerification(t *testing.T) {
+	log, buf := upgrade.NewInMemoryLoggerBuffer()
 	skipMsg := "It is expected to be skipped"
 	s := upgrade.Suite{
 		Tests: upgrade.Tests{
 			Continual: []upgrade.BackgroundOperation{
 				upgrade.NewBackgroundVerification("ShouldBeSkipped",
 					func(c upgrade.Context) {
-						c.Log.Info("Setup 1")
+						log.Info("Setup 1")
 					},
 					func(c upgrade.Context) {
-						c.Log.Warn(skipMsg)
+						log.Warn(skipMsg)
 						c.T.Skip(skipMsg)
 					},
 				),
 				upgrade.NewBackgroundVerification("ShouldNotBeSkipped",
 					func(c upgrade.Context) {
-						c.Log.Info("Setup 2")
+						log.Info("Setup 2")
 					},
 					func(c upgrade.Context) {
-						c.Log.Info("Verify 2")
+						log.Info("Verify 2")
 					},
 				),
 			},
 		},
 	}
-	log, buf := newExampleZap()
```

Comment:
I don't see a valid point why those changes are needed here in this test. Now we are logging, no matter if test fail or success. This test should work as it was set up before.

Reply:
We print the output to t.Log in the end but the test was checking that the original in memory logger had the logs which is no longer true. The internal logger has the logs and they're printed to t.Log

Reply:
I believe the point of this test is to check what is and is not executed (it's about skipping the test). And for this purpose it's enough to pass a separate logger like I'm going with this change.